src/components/join-flow/gender-step.jsx

```diff
@@ -0,0 +1,145 @@
+const bindAll = require('lodash.bindall');
+const classNames = require('classnames');
+const React = require('react');
+const PropTypes = require('prop-types');
+import {Formik} from 'formik';
+const {injectIntl, intlShape} = require('react-intl');
+
+const FormikRadioButton = require('../../components/formik-forms/formik-radio-button.jsx');
+const JoinFlowStep = require('./join-flow-step.jsx');
+
+require('./join-flow-steps.scss');
+
+const GenderOption = ({
+    label,
+    onSetFieldValue,
+    selectedValue,
+    value
+}) => (
+    <div
+        className={classNames(
+            'col-sm-9',
+            'row',
+            'row-inline',
+            'gender-radio-row',
+            {'gender-radio-row-selected': (selectedValue === value)}
+        )}
+        /* eslint-disable react/jsx-no-bind */
+        onClick={() => onSetFieldValue('gender', value, false)}
+        /* eslint-enable react/jsx-no-bind */
+    >
+        <FormikRadioButton
+            buttonValue={value}
+            className={classNames(
+                'join-flow-radio'
+            )}
+            label={label}
+            name="gender"
+        />
+    </div>
+);
+
+GenderOption.propTypes = {
+    label: PropTypes.string,
+    onSetFieldValue: PropTypes.func,
+    selectedValue: PropTypes.string,
+    value: PropTypes.string
+};
+
+class GenderStep extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleValidSubmit'
+        ]);
+    }
+    handleValidSubmit (formData, formikBag) {
+        formikBag.setSubmitting(false);
+        if (!formData.gender || formData.gender === 'null') {
+            formData.gender = 'Prefer not to say';
+        }
+        delete formData.other;
+        this.props.onNextStep(formData);
+    }
+    render () {
+        return (
+            <Formik
+                initialValues={{
+                    gender: 'null',
+                    other: ''
+                }}
+                onSubmit={this.handleValidSubmit}
+            >
+                {props => {
+                    const {
+                        handleSubmit,
+                        isSubmitting,
+                        setFieldValue,
```

Comment:
Silly React newbie question: Where is setFieldValue sent in here?  i.e.  below in each <GenderOption>, there's the onSetFieldValue={setFieldValue} line. Where is the actual definition  of that?

Reply:
Short answer: `<Formik>` assigns a `props` object, which we use to unpack whichever values we want to use. The possible values we can use are listed here: https://jaredpalmer.com/formik/docs/api/formik

Long answer: This is one of the magical (I mean that in a bad way) things that js can do, that formik takes advantage of. Basically, when we say `<Formik>{some stuff here}</Formik>`, react passes the `{some stuff here}` to the `Formik` component as the `children` prop. As I (shakily) understand it, `Formik` expects `children` to be a function, and in its render function it calls `{children({props object here})}`. So the `Formik` component actually renders all this `{some stuff here}`, NOT the `GenderStep` component itself.

Confusing!